packages/api/db/migration/20240111141747_create_animal_breed_table.js

```diff
@@ -0,0 +1,103 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+export const up = async function (knex) {
+  await knex.schema.createTable('animal_breed', (table) => {
+    table.increments('id').primary();
```

Comment:
I know I suggested the increments as a primary key. Just wondering where we landed on the increments per farm. so making the primars key unique to the farm, and mayyyybe the breed as well . (ie many with id = 1, but only one id = 1 && farm_id = xxxx)

Reply:
Also wondering what you thought of renaming to `animal_breed_id`, id makes sense for the context of table but I am guessing the reason we are explicit is we usually work with multiple tables and if they all used 'id' in the context of the table it might get confusing.

Reply:
The key set to autoincrement would still be unique per farm, so IMo I'd just let Postgres handle it, as long as they don't repeat each other it's all good. I'm not sure how we'd handle making them unique per farm, would we use the farm ID as some sort of seed? Seems a bit overcomplicated IMO.

For the ids, I was hesitant about it but that is the pattern that we follow on other tables and I can see how it could be useful when doing joins to avoid having to rename the columns, so I'll update it!

Reply:
Primary key: I guess I am stuck on the concept of farm specific increments like how Jira/ Github Issue starts off as 1 for every organization. Its true it doesn't apply to all situations -- short urls is nice tho. But I think its maybe easy to make a composite key https://stackoverflow.com/a/56744896 . If I think harder it is a half solution to ideal anyways so yah it doesn't matter... but maybe the perfect ideal result would be readable URLs like `/animal_breed/cow/ `vs `/animal_breed/22/ `

ID: Yah I like the way you have it better I am trying to understand why maybe it was done the way it is and if consistency is ok if there is a good enough reason

Reply:
Yeah I think the key here is there's a difference in what's shown to the user and what's used as a primary key for the table -- Jira has a company identifier (like "LF") and a ticket number which they extract from the URL and use to lookup the right ticket int their database. This doesn't mean though that they don't have an internal ID that is used as a primary key. We could definitely do a more user friendly URL if we want to, but I would add extra fields to do that rather than use our primary key ID.

Reusing the same id for different farms by using a composite key with the farm ID could end up being error prone as we'd need to look at the farm ID to distinguish one record from the other, and use both fields to lookup a record.

Reply:
Good point, yah unfortunately we do use primary key on FE url.